airbyte-integrations/bases/base-normalization/unit_tests/test_destination_name_transformer.py

```diff
@@ -171,6 +171,7 @@ def test_normalize_column_name(input_str: str, destination_type: str, expected:
         ("Aaaa_Bbbb_Cccc_Dddd_Eeee_Ffff_Gggg_Hhhh_Iii", "Aaaa_Bbbb_Cccc_Dddd_Eeee_Ffff_Gggg_Hhhh_Iii"),
         # over the limit
         ("Aaaa_Bbbb_Cccc_Dddd_Eeee_Ffff_Gggg_Hhhh_Iiii", "Aaaa_Bbbb_Cccc_Dddd___e_Ffff_Gggg_Hhhh_Iiii"),
+        ("Aaaa_Bbbb_Cccc_Dddd_a_very_long_name_Ffff_Gggg_Hhhh_Iiii", "Aaaa_Bbbb_Cccc_Dddd___e_Ffff_Gggg_Hhhh_Iiii"),
```

Comment:
When truncated, this test case ends up with the exact same truncated name as the previous test... possible conflict is possible

Reply:
nice catch! that's bad. how should we handle it?

Reply:
it looks like in the implementation above you're throwing an error. is there anyway to handle this without throwing an error?

Reply:
when it's finding a duplicate name, the code tries a second chance by adding a 3 character hash of the full stream name (without truncation) at the end of the truncated stream name.

If that second chance still has collisions (because the 3 character hash was already used too? or maybe because the catalog somehow contains the exact same stream name twice) then it'll fail 

We could make it not throw an error by appending a random suffix character or number (_1, _2, _3 etc) instead of erroring but it's not deterministic anymore ...